source/fe/fe_enriched.cc

```diff
@@ -0,0 +1,949 @@
+                                          const Mapping<dim,spacedim>    &mapping,
+                                          const Quadrature<dim-1> &quadrature,
+                                          internal::FEValues::FiniteElementRelatedData< dim, spacedim >        &output_data) const
+{
+  return setup_data(fe_system.get_face_data(update_flags,mapping,quadrature,output_data),
+                    update_flags,
+                    mapping,
+                    quadrature,
+                    output_data,
+                    false);
+}
+
+
+template <int dim, int spacedim>
+typename FiniteElement<dim,spacedim>::InternalDataBase *
+FE_Enriched<dim,spacedim>::get_subface_data (const UpdateFlags      update_flags,
+                                             const Mapping<dim,spacedim>    &mapping,
+                                             const Quadrature<dim-1> &quadrature,
+                                             dealii::internal::FEValues::FiniteElementRelatedData<dim, spacedim> &output_data) const
+{
+  return setup_data(fe_system.get_subface_data(update_flags,mapping,quadrature,output_data),
+                    update_flags,
+                    mapping,
+                    quadrature,
+                    output_data,
+                    true);
+}
+
+
+template <int dim, int spacedim>
+typename FiniteElement<dim,spacedim>::InternalDataBase *
+FE_Enriched<dim,spacedim>::get_data (const UpdateFlags      flags,
+                                     const Mapping<dim,spacedim>    &mapping,
+                                     const Quadrature<dim> &quadrature,
+                                     internal::FEValues::FiniteElementRelatedData< dim, spacedim >   &output_data) const
+{
+  return setup_data(fe_system.get_data(flags,mapping,quadrature,output_data),
+                    flags,
+                    mapping,
+                    quadrature,
+                    output_data,
+                    false);
+}
+
+
+template <int dim, int spacedim>
+FE_Enriched<dim,spacedim>::FE_Enriched (const FiniteElement<dim,spacedim> &fe_base)
+  :
+  FE_Enriched<dim,spacedim>(fe_base,
+                            FE_Nothing<dim,spacedim>(fe_base.n_components(),true),
+                            NULL)
+{
+}
+
+
+template <int dim, int spacedim>
+FE_Enriched<dim,spacedim>::FE_Enriched (const FiniteElement<dim,spacedim> &fe_base,
+                                        const FiniteElement<dim,spacedim> &fe_enriched,
+                                        const Function<dim>      *enrichment_function)
+  :
+  FE_Enriched<dim,spacedim>
+  (&fe_base, { &fe_enriched },
+{
+  {
+    [=] (const typename Triangulation<dim, spacedim>::cell_iterator &) -> const Function<dim> *
+    {
+      return enrichment_function;
+    }
+  }
+})
+{
+}
+
+
+template <int dim, int spacedim>
+FE_Enriched<dim,spacedim>::FE_Enriched (const FiniteElement<dim,spacedim> *fe_base,
```

Comment:
One could do that, but i would still need to have `fe_enriched` to be a vector of pointers because eventually i call the most general constructor of `FESystem` with `const std::vector<const FiniteElement<dim,spacedim>*> &fes,`. 

To make the two input arguments consistent, i would prefer to keep `fe_base` as a pointer.